ios/PlayerModule.swift

```diff
@@ -662,4 +666,16 @@ class PlayerModule: NSObject, RCTBridgeModule {
             self?.players[nativeId]?.castStop()
         }
     }
+
+    private func setupRemoteControlConfig(_ remoteControlConfig: RemoteControlConfig) {
+        remoteControlConfig.prepareSource = { [weak self] type, sourceConfig in
+            guard let sourceModule = self?.bridge[SourceModule.self],
+                  let sourceNativeId = sourceModule.nativeId(where: { $0.sourceConfig === sourceConfig }),
+                  let castSourceConfig = sourceModule.retrieveCastSourceConfig(sourceNativeId) else {
+                return nil
+            }
+
+            return castSourceConfig
```

Comment:
We have to look up the cast-specific source config on the fly and we have to use `sourceConfig` to do that, as `player.source` returns `nil` when this is called.

Reply:
@strangesource `===` is not equality check but identity check. Therefore it would need to be the same object instance (same pointer/memory address).
This would be a problem for a fully native project as you can create multiple sources with the same `SourceConfig` instance, BUT since in React Native we always parse the JSON and create a new `SourceConfig` object, the reuse of the same instance of `SourceConfig` can not happen.

Reply:
Thanks for the explanation. This should be correct then. It does require some deeper understanding of the domain to understand though - which is fine I guess.